tests/FSharp.Compiler.ComponentTests/Conformance/Types/TypeConstraints/IWSAMsAndSRTPs/IWSAMsAndSRTPsTests.fs

```diff
@@ -1101,7 +1101,143 @@ module StaticAbstractBug =
+         ]
+         
+    [<FactForNETCOREAPP>]
+    let ``Produce an error when one leaves keyword "static" when implementing static and not static members in an object expression`` () =
+        Fsx """
+type IOperation =
+        static abstract member Execute: unit -> unit
+        abstract member Execute1: unit -> unit
+        static abstract member Execute2: unit -> unit
+        abstract member Execute3: bool
+        
+let _ =
+    { new IOperation with
+        static member Execute() = ()
+        member this.Execute1() = ()
+        static member Execute2() = ()
+        member this.Execute3 = true
+    }
+        """
+         |> withOptions [ "--nowarn:3536" ; "--nowarn:3535" ]
+         |> withLangVersion80
+         |> typecheck
+         |> shouldFail
+         |> withDiagnostics [
+            (Error 3860, Line 10, Col 23, Line 10, Col 30, "Static members are not allowed in object expressions.");
+            (Error 3860, Line 12, Col 23, Line 12, Col 31, "Static members are not allowed in object expressions.");
+         ]
          
+    [<FactForNETCOREAPP>]
+    let ``Produces an error when implementing only instance members from IWSAM in an object expression`` () =
+        Fsx """
+type ILogger =
+    abstract member Log: string -> unit
+    static abstract member Execute: string -> unit
+        
+let consoleLogger =
+    { new ILogger with
+        member this.Log(message: string) =
+            printfn "%s" message
+    }
+    
+consoleLogger.Log("Hello World")
+        """
+         |> withOptions [ "--nowarn:3536" ; "--nowarn:3535" ]
+         |> withLangVersion80
+         |> compile
+         |> shouldFail
+         |> withDiagnostics [
+             (Error 366, Line 7, Col 5, Line 10, Col 6, "No implementation was given for 'static abstract ILogger.Execute: string -> unit'. Note that all interface members must be implemented and listed under an appropriate 'interface' declaration, e.g. 'interface ... with member ...'.")
+         ]
+         
+    [<FactForNETCOREAPP>]
+    let ``Produces an error when implementing only instance members from IWSAM(Interface attribute) in an object expression`` () =
+        Fsx """
+[<Interface>]
+type ILogger =
+    abstract member Log: string -> unit
+    static abstract member Execute: string -> unit
+        
+let consoleLogger =
+    { new ILogger with
+        member this.Log(message: string) =
+            printfn "%s" message
+    }
+    
+consoleLogger.Log("Hello World")
+        """
+         |> withOptions [ "--nowarn:3536" ; "--nowarn:3535" ]
+         |> withLangVersion80
+         |> compile
+         |> shouldFail
+         |> withDiagnostics [
+             (Error 366, Line 8, Col 5, Line 11, Col 6, "No implementation was given for 'static abstract ILogger.Execute: string -> unit'. Note that all interface members must be implemented and listed under an appropriate 'interface' declaration, e.g. 'interface ... with member ...'.")
```

Comment:
I think this should work in general, but it's a different PR probably.

Reply:
Happy to create a follow up PR about not producing diagnostic about static member in object expression.